pkg/reconciler/kubernetes/tektonresult/tektonresult.go

```diff
@@ -229,6 +229,8 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tr *v1alpha1.TektonResul
 				return err
 			}
 
+			r.manifest = r.manifest.Filter(filterExternalDB(tr.Spec.IsExternalDB))
```

Comment:
Yes, we could done something like this but it's not that much efficient. We have three context calls instead of one when we filter but it won't happen unless the code arrives there.

Another thing to note is we lose on unit testing here because we aren't unit testing ReconcileKind. I would prefer to have a separate function in ReconcileKind so that we can unit test.  @jkandasa

Reply:
So instead of 			
```
if tr.Spec.IsExternalDB {
	 r.manifest = r.manifest.Filter(mf.Not(mf.All(mf.ByKind("StatefulSet"), mf.ByName("tekton-results-postgres"))))
}
```
 We could have 
 ```
 r. manifest = filterExternalDB(r.manifest)
 ```
 And `filterExternalDB` can have those lines.